src/Microsoft.NetCore.Analyzers/Core/Security/SourceTriggeredTaintedDataAnalyzerBase.cs

```diff
@@ -77,62 +85,164 @@ public override void Initialize(AnalysisContext context)
-                                        rootOperationsNeedingAnalysis.Add(operationAnalysisContext.Operation.GetRoot());
+                                        IOperation rootOperation = operationAnalysisContext.Operation.GetRoot();
+                                        PointsToAnalysisResult pointsToAnalysisResult;
+                                        ValueContentAnalysisResult valueContentAnalysisResultOpt;
+                                        if (evaluateWithPointsToAnalysis != null)
+                                        {
+                                            pointsToAnalysisResult = PointsToAnalysis.TryGetOrComputeResult(
+                                                rootOperation.GetEnclosingControlFlowGraph(),
+                                                owningSymbol,
+                                                WellKnownTypeProvider.GetOrCreate(operationAnalysisContext.Compilation),
+                                                InterproceduralAnalysisConfiguration.Create(
+                                                    operationAnalysisContext.Options,
+                                                    SupportedDiagnostics,
+                                                    defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.ContextSensitive,
+                                                    cancellationToken: operationAnalysisContext.CancellationToken),
+                                                interproceduralAnalysisPredicateOpt: null);
+                                            if (pointsToAnalysisResult == null)
+                                            {
+                                                return;
+                                            }
+
+                                            try
+                                            {
+                                                if (evaluateWithPointsToAnalysis.Any(s => s(
+                                                    invocationOperation.Arguments.Select(o => pointsToAnalysisResult[o.Kind, o.Syntax]))))
+                                                {
+                                                    lock (rootOperationsNeedingAnalysis)
+                                                    {
+                                                        rootOperationsNeedingAnalysis.Add(rootOperation);
+                                                    }
+                                                }
+                                            }
+                                            finally
+                                            {
+                                                evaluateWithPointsToAnalysis.Free();
+                                            }
+                                        }
+                                        else if (evaluateWithValueContentAnalysis != null)
+                                        {
```

Comment:
Are evaluateWithPointsToAnalysis and evaluateWithValueContentAnalysis mutually exclusive?  Is it possible to for the same method name to have callbacks for both?

It doesn't look like this code handles that case.  Same in the operation visitor code.

Reply:
1) If a method needs to do value content and points to check at the same time, it will be put in the taintedMethodsNeedsValueContentAnalysis set.
2) No, they are not mutually exclusive. But any method meets one requirement in taintedMethodsNeedPointsToAnalysis or taintedMethodsNeedsValueContentAnalysis is tainted, that's by design.